qtfred/src/mission/missionsave.cpp

```diff
@@ -2120,6 +2121,200 @@ int CFred_mission_save::save_messages()
 	return err;
 }
 
+int CFred_mission_save::save_campaign_file(const char *pathname)
+{
+	reset_parse();
+	fred_parse_flag = 0;
+
+	pathname = cf_add_ext(pathname, FS_CAMPAIGN_FILE_EXT);
+	fp = cfopen(pathname, "wt", CFILE_NORMAL, CF_TYPE_MISSIONS);
+	if (!fp) {
+		nprintf(("Error", "Can't open campaign file to save.\n"));
+		return -1;
+	}
+
+	required_string_fred("$Name:");
+	parse_comments(0);
+	fout(" %s", Campaign.name);
+
+	Assert((Campaign.type >= 0) && (Campaign.type < MAX_CAMPAIGN_TYPES));
+	required_string_fred("$Type:");
+	parse_comments();
+	fout(" %s", campaign_types[Campaign.type]);
+
+	// XSTR
+	if (Campaign.desc) {
+		required_string_fred("+Description:");
+		parse_comments();
+		fout_ext("\n", "%s", Campaign.desc);
+		fout("\n$end_multi_text");
+	}
+
+	if (Campaign.type != CAMPAIGN_TYPE_SINGLE) {
+		required_string_fred("+Num Players:");
+		parse_comments();
+		fout(" %d", Campaign.num_players);
+	}
+
+	// campaign flags - Goober5000
+	if (save_format != MissionFormat::RETAIL) {
+		optional_string_fred("$Flags:");
+		parse_comments();
+		fout(" %d\n", Campaign.flags);
+	}
+
+	// write out the ships and weapons which the player can start the campaign with
+	optional_string_fred("+Starting Ships: (");
+	parse_comments(2);
+	for (int i = 0; i < ship_info_size(); i++) {
+		if (Campaign.ships_allowed[i])
```

Comment:
Please be consistent with the existing single-line `if` statement's brace style in this file:  Add a `{ }`